packages/ui/src/pages/Register/index.tsx

```diff
@@ -1,63 +1,63 @@
-import React, { FC, FormEventHandler, useState, useCallback } from 'react';
+import React, { FC, FormEventHandler, useState, useCallback, useEffect } from 'react';
 import { useTranslation } from 'react-i18next';
 import classNames from 'classnames';
+import { LogtoErrorI18nKey } from '@logto/phrases';
 
 import { register } from '@/apis/register';
 import Button from '@/components/Button';
 import Input from '@/components/Input';
 import MessageBox from '@/components/MessageBox';
 import TextLink from '@/components/TextLink';
+import useApi from '@/hooks/use-api';
 
 import styles from './index.module.scss';
 
-export type PageState = 'idle' | 'loading' | 'error';
-
 const App: FC = () => {
-  const { t } = useTranslation();
+  const { t, i18n } = useTranslation();
   const [username, setUsername] = useState('');
   const [password, setPassword] = useState('');
-  const [pageState, setPageState] = useState<PageState>('idle');
-  const isLoading = pageState === 'loading';
+
+  const { loading, error, result, fetch: fireRegister } = useApi(register);
```

Comment:
should be good with `register` or something simpler? fire + verb sounds a little bit strange (register could be a noun, but it's verb here i think)

Reply:
updated